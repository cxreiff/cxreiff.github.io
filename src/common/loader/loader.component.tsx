import { FC } from 'react'

/** TODO: Change to `import * as styles from...` to re-enable tree shaking
*  Currently disabling tree shaking for this import because of issue where
*  keyframes definition is removed in production builds.
*/
import styles from './loader.module.scss'

export const Loader: FC = () => (
    <div className={styles.loader}>
        {[1,2,3,4].map((key) => (
            <span key={key} />
        ))}
    </div>
)
