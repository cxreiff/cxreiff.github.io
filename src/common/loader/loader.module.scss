@use '../../shared.scss' as *;

$num-colors: 4;

.loader {

    margin: 2rem 1rem 1rem;
    text-align: center;

    span {
        display: inline-block;
        width: 1rem;
        height: 2rem;
        margin: 0.25rem;
        opacity: 0.3;
    }

    @for $i from 0 to $num-colors {
        span:nth-child(#{$num-colors}n + #{$i + 1}) {
            background: $accent-color-dark;
            animation: fade-pulse ease-in-out 1.5s infinite;
            animation-delay: $i * 250ms;
        }
    }
}

@keyframes fade-pulse {
    0% {
        opacity: 0.3;
    background: $accent-color-dark;
    }

    40% {
        opacity: 1;
    background: $accent-color-tertiary;
    }

    80% {
        opacity: 0.3;
    background: $accent-color-dark;
    }
}
