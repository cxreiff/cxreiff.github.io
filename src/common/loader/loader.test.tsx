import { mount, ReactWrapper } from 'enzyme'

import { Loader } from './loader.component'

describe('loader', () => {

  let component: ReactWrapper<typeof Loader, void>

  beforeEach(() => {
    component = mount<typeof Loader, void>(
      <Loader />
    )
  })

  afterEach(() => {
    component.unmount()
  })

  it('should render', () => {
    expect(component).toMatchSnapshot()
  })
})
