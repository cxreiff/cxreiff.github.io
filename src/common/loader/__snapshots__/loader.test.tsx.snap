// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`loader should render 1`] = `
<div
  className="loader"
>
  <span
    key={1}
  />
  <span
    key={2}
  />
  <span
    key={3}
  />
  <span
    key={4}
  />
</div>
`;
